components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/util/WebhookValidator.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.webhook.management.internal.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.webhook.management.api.exception.WebhookMgtClientException;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static org.wso2.carbon.identity.webhook.management.api.constant.ErrorMessage.ERROR_EMPTY_WEBHOOK_REQUEST_FIELD;
+import static org.wso2.carbon.identity.webhook.management.api.constant.ErrorMessage.ERROR_INVALID_WEBHOOK_REQUEST_FIELD;
+import static org.wso2.carbon.identity.webhook.management.internal.constant.WebhookMgtConstants.CHANNELS_SUBSCRIBED_FIELD;
+import static org.wso2.carbon.identity.webhook.management.internal.constant.WebhookMgtConstants.ENDPOINT_URI_FIELD;
+import static org.wso2.carbon.identity.webhook.management.internal.constant.WebhookMgtConstants.WEBHOOK_NAME_FIELD;
+
+/**
+ * Webhook validator class.
+ */
+public class WebhookValidator {
+
+    private static final String WEBHOOK_NAME_REGEX = "^[a-zA-Z0-9-_][a-zA-Z0-9-_ ]*[a-zA-Z0-9-_]$";
+    private static final String ENDPOINT_URI_REGEX = "^https?://[^\\s/$.?#]\\S*";
+    // According to RFC 9910 a header name must contain only alphanumeric characters, period (.) and hyphen (-),
+    // and should start with an alphanumeric character.
+    private static final String SECRET_REGEX = "^[A-Za-z0-9._~+-]{32,128}$";
```

Comment:
I think we should support a more character set than this.
I got following suggested for all printable ascii characters
^[\x20-\x7E]+$

\x20 is the space character (ASCII 32).
\x7E is the tilde (~) character (ASCII 126).

This range includes:

Digits: 0-9
Uppercase letters: A-Z
Lowercase letters: a-z
Symbols and punctuation: !"#$%&'()*+,-./:;<=>?@[\]^_{|}~
Space ( )

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/6821/commits/97cfe8e2faea721aac846e5187919e3950c58c75